### PR TITLE
Cleanup / type MathInput (pt 7)

### PR DESCRIPTION
**Pull request by handeyeco**
## Summary:
Hopefully the final cleanup

- switch `Array` to `ReadonlyArray` when possible
- delete unused prop types

Issue: LC-740

## Test plan:

### CODE REVIEW

File: packages/math-input/src/components/keypad/button.tsx [hunk 1]
```diff
@@ -65,8 +65,7 @@ type Props = {
     tintColor?: string;
 };
 
-type State = Record<any, any>;
```
Comment (handeyeco) [inline packages/math-input/src/components/keypad/button.tsx:68]: I didn't see the point in having these if there was no state.

File: packages/math-input/src/index.ts [hunk 2]
```diff
@@ -5,10 +5,7 @@
 import "../less/main.less";
 
 export {default as KeypadInput} from "./components/input/math-input";
-export {
-    keypadConfigurationPropType,
-    keypadElementPropType,
-} from "./components/prop-types";
+export {keypadElementPropType} from "./components/prop-types";
```
Comment (handeyeco) [inline packages/math-input/src/index.ts:8]: API change. Should this be a major/minor changeset?

File: packages/math-input/src/components/icon.tsx [hunk 3]
```diff
@@ -28,17 +28,11 @@ class Icon extends React.PureComponent<Props> {
     render() {
         const {focused, icon, style} = this.props;
 
-        let styleWithFocus: StyleType = [
+        const styleWithFocus: StyleType = [
             focused ? styles.focused : styles.unfocused,
+            ...(Array.isArray(style) ? style : [style]),
```
Comment (jeremywiebe) [inline packages/math-input/src/components/icon.tsx:33]: Thanks!

File: packages/math-input/src/components/input/math-input.tsx [hunk 4]
```diff
@@ -13,18 +13,21 @@ import {
...
+import type {Cursor} from "../../types";
+
 const constrainingFrictionFactor = 0.8;
 
 type Props = {
-    keypadElement: any;
+    keypadElement: ProvidedKeypad;
     onBlur: () => void;
-    onChange: any;
+    onChange: (value: string, cb: any) => void;
```
Comment (jeremywiebe) [inline packages/math-input/src/components/input/math-input.tsx:30]: One... tiny... request.  Does it work if you define `cb` as:
```suggestion
    onChange: (value: string, callback: (any) => void) => void;
```
This'll help readers know it's a function and not something else.  Also, the naming is a bit obscure, so maybe `callback` instead?

File: packages/math-input/src/components/input/math-wrapper.ts [hunk 5]
```diff
@@ -119,8 +119,8 @@ const KeysForJumpContext = {
 };
 
 class MathWrapper {
-    MQ: any;
-    mathField: any;
+    MQ: any; // MathQuill interface
+    mathField: any; // MathQuill input
```
Comment (jeremywiebe) [inline packages/math-input/src/components/input/math-wrapper.ts:123]: 🔥 Nice!

File: packages/math-input/src/components/many-keypad-button.tsx [hunk 6]
```diff
@@ -15,11 +15,12 @@ import TouchableKeypadButton from "./touchable-keypad-button";
 import type {Key} from "../data/keys";
 
 type Props = {
-    keys?: Array<Key>;
+    keys?: ReadonlyArray<Key>;
 };
 
 class ManyKeypadButton extends React.Component<Props> {
     render() {
+        // do we even need the ...rest?
```
Comment (jeremywiebe) [inline packages/math-input/src/components/many-keypad-button.tsx:23]: Don't we use it on line 33 and 40?

File: packages/math-input/src/components/prop-types.js
Comment (jeremywiebe) [inline packages/math-input/src/components/prop-types.js:1]: Love all the deletions here! Thank-you!

File: packages/math-input/src/components/provided-keypad.tsx [hunk 7]
```diff
@@ -14,17 +14,17 @@ import {createStore} from "../store/index";
...
     onElementMounted?: (arg1: any) => void;
-    onDismiss?: () => unknown;
+    onDismiss?: () => void;
     style?: StyleType;
 };
 
 class ProvidedKeypad extends React.Component<Props> {
-    // @ts-expect-error [FEI-5003] - TS2564 - Property 'mounted' has no initializer and is not definitely assigned in the constructor.
-    mounted: boolean;
+    mounted?: boolean;
```
Comment (jeremywiebe) [inline packages/math-input/src/components/provided-keypad.tsx:27]: **nit**: I am guessing our code doesn't care between `undefined` and `false` so we can avoid initializing it, but I feel its slightly more accurate/clear to initialize it here and then make it non-optional. 

```suggestion
    mounted: boolean = false;
```

File: packages/math-input/src/index.ts (hunk 2 above)
Comment (jeremywiebe) [reply packages/math-input/src/index.ts:8]: Major, but I asked https://khanacademy.slack.com/archives/CEA6W0F6F/p1681488884159309

**Review by jeremywiebe — APPROVED** [review]
Excellent.  Let's just wait to figure out the server (major/minor version bump) question before we land this.

File: packages/math-input/src/index.ts (hunk 2 above)
Comment (jeremywiebe) [reply packages/math-input/src/index.ts:8]: And the answer is: it's a major version bump!

File: packages/math-input/src/components/input/math-input.tsx (hunk 4 above)
Comment (handeyeco) [reply packages/math-input/src/components/input/math-input.tsx]: I'll switch the name, but the type is problematic. Anytime I touch it, errors start to spread. For instance somewhere in the same file they're setting it to a boolean.

File: packages/math-input/src/components/input/math-wrapper.ts (hunk 5 above)
Comment (handeyeco) [reply packages/math-input/src/components/input/math-wrapper.ts:123]: I tried to change the `mathField` name but it broke everything 😭

File: packages/math-input/src/components/many-keypad-button.tsx (hunk 6 above)
Comment (handeyeco) [reply packages/math-input/src/components/many-keypad-button.tsx:23]: Yes, but TouchableKeypadButton is kind of a mess and it's a little difficult for me to parse what it actually needs. https://khanacademy.atlassian.net/browse/LC-746

Comment (handeyeco) [reply packages/math-input/src/components/many-keypad-button.tsx]: Removed

**Pull request merged**